storage/entities/transaction.js

```diff
@@ -402,12 +418,10 @@ class Transaction extends BaseEntity {
 		switch (type) {
 			case 0:
 				fields = ['transactionId', 'data'];
-				values = transactions
-					.filter(transaction => transaction.asset && transaction.asset.data)
-					.map(transaction => ({
-						transactionId: transaction.id,
-						data: Buffer.from(transaction.asset.data, 'utf8'),
-					}));
+				values = transactions.map(transaction => ({
+					transactionId: transaction.id,
+					data: Buffer.from(transaction.asset.data, 'utf8'),
+				}));
```

Comment:
Something like this. 

```
values = [];
transactions.forEach(transaction => {
 if(any condition)
values.push({})
});
```

Idea is the transaction type based any logic should be scoped on one place. Later we may extract it to create seperate transaction type classes, or different function for each type.

Reply:
Updated based on suggestion.